app/scripts/background.js

```diff
@@ -603,6 +601,9 @@ async function initialize(backup) {
 
   const preinstalledSnaps = await loadPreinstalledSnaps();
 
+  const { update, requestSafeReload } =
+    getRequestSafeReload(persistenceManager);
```

Comment:
This ends up wrapping the `persistenceManager.set` call in, essentially in this `update` function.